st2debug/st2debug/utils/system_info.py

```diff
@@ -0,0 +1,191 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+import pipes
+
+from st2common.util.shell import run_command
+
+__all__ = [
+    'get_cpu_info',
+    'get_memory_info',
+    'get_package_list',
+    'get_deb_package_list',
+    'get_rpm_package_list'
+]
+
+CPU_INFO_PATH = '/proc/cpuinfo'
```

Comment:
Yeah, there are quite a lot of existing packages for doing that, but most of them (e.g. sigar) have C dependencies which I'd like to avoid.

Will look around some more.